src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/Legacy/RoslynCSharpTokenizer.cs

```diff
@@ -1,173 +1,101 @@
-        _operatorHandlers = new Dictionary<char, Func<SyntaxKind>>()
-            {
-                { '-', MinusOperator },
-                { '<', LessThanOperator },
-                { '>', GreaterThanOperator },
-                { '&', CreateTwoCharOperatorHandler(SyntaxKind.And, '=', SyntaxKind.AndAssign, '&', SyntaxKind.DoubleAnd) },
-                { '|', CreateTwoCharOperatorHandler(SyntaxKind.Or, '=', SyntaxKind.OrAssign, '|', SyntaxKind.DoubleOr) },
-                { '+', CreateTwoCharOperatorHandler(SyntaxKind.Plus, '=', SyntaxKind.PlusAssign, '+', SyntaxKind.Increment) },
-                { '=', CreateTwoCharOperatorHandler(SyntaxKind.Assign, '=', SyntaxKind.Equals, '>', SyntaxKind.GreaterThanEqual) },
-                { '!', CreateTwoCharOperatorHandler(SyntaxKind.Not, '=', SyntaxKind.NotEqual) },
-                { '%', CreateTwoCharOperatorHandler(SyntaxKind.Modulo, '=', SyntaxKind.ModuloAssign) },
-                { '*', CreateTwoCharOperatorHandler(SyntaxKind.Star, '=', SyntaxKind.MultiplyAssign) },
-                { ':', CreateTwoCharOperatorHandler(SyntaxKind.Colon, ':', SyntaxKind.DoubleColon) },
-                { '?', CreateTwoCharOperatorHandler(SyntaxKind.QuestionMark, '?', SyntaxKind.NullCoalesce) },
-                { '^', CreateTwoCharOperatorHandler(SyntaxKind.Xor, '=', SyntaxKind.XorAssign) },
-                { '(', () => SyntaxKind.LeftParenthesis },
-                { ')', () => SyntaxKind.RightParenthesis },
-                { '{', () => SyntaxKind.LeftBrace },
-                { '}', () => SyntaxKind.RightBrace },
-                { '[', () => SyntaxKind.LeftBracket },
-                { ']', () => SyntaxKind.RightBracket },
-                { ',', () => SyntaxKind.Comma },
-                { ';', () => SyntaxKind.Semicolon },
-                { '~', () => SyntaxKind.Tilde },
-                { '#', () => SyntaxKind.Hash }
-            };
+        _roslynTokenParser = CodeAnalysis.CSharp.SyntaxFactory.CreateTokenParser(source.SourceText, null);
     }
 
-    protected override int StartState => (int)CSharpTokenizerState.Data;
+    protected override int StartState => (int)RoslynCSharpTokenizerState.Start;
 
-    private new CSharpTokenizerState? CurrentState => (CSharpTokenizerState?)base.CurrentState;
+    private new RoslynCSharpTokenizerState? CurrentState => (RoslynCSharpTokenizerState?)base.CurrentState;
 
     public override SyntaxKind RazorCommentKind => SyntaxKind.RazorCommentLiteral;
 
     public override SyntaxKind RazorCommentTransitionKind => SyntaxKind.RazorCommentTransition;
 
     public override SyntaxKind RazorCommentStarKind => SyntaxKind.RazorCommentStar;
 
+    internal override void StartingBlock()
+    {
+        _roslynTokenParser.SkipForwardTo(Source.Position);
+    }
+
+    internal override void EndingBlock()
+    {
+        // We should always be transitioning to the other parser in response to content. This means that we'll have parsed a token, and be in the TriviaForCSharpToken
```

Comment:
Consider replacing "parser" with "tokenizer" in this comment.

Reply:
I actually meant parser, not tokenizer.